### PR TITLE
Issue #2931731 by tjwelde: [UX] Field length indicator without setting a hard limit

### PR DESCRIPTION
**Pull request by goldquest**
Make sure these boxes are checked before submitting your pull request - thank you!

- [x] All coding styles are fulfilled. ([How to check for cs issues?](https://github.com/BurdaMagazinOrg/thunder-dev-tools/blob/master/README.md#code-style-guidelines))
- [x] All tests are running locally. ([How to run the test?](https://github.com/BurdaMagazinOrg/thunder-distribution/blob/develop/docs/development.md#how-to-run-the-tests))
- [x] Necessary update hooks are provided.
- [x] User roles have correct access for new introduced permission.
- [x] Every thunder module has a README.md in its root. Follow [this guidelines](https://www.drupal.org/node/2181737), but we don't need every topic.
- [x] Code is covered with well-balanced amount of inline comments.

If you are really awesome, then your feature is covered by additional tests. Well done!

### CODE REVIEW

**Comment by chrfritsch** [comment]
Thats pretty much code and i'm glad that it's not coupled to Thunder. So please create a d.o. project for that.

File: modules/thunder_article/modules/thunder_indicator/thunder_indicator.module [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+ *   The optimum minimum.
+ * @param int $optimax
+ *   The optimum maximum.
+ * @param int $tolerance
+ *   The tolerance.
+ *
+ * @return array
+ *   The widths and positions of the indicators.
+ */
+function _thunder_indicator_get_width_and_pos($optimin, $optimax, $tolerance) {
```
Comment (chrfritsch) [inline modules/thunder_article/modules/thunder_indicator/thunder_indicator.module]: I think this should be moved into a service

**Comment by mtodor** [comment]
I have tested this a bit, looks good from my point of view. There are 2 things that I'm not sure of:

1. default numbers for teaser - they are really small and if you base your teaser content on length indicator then teaser doesn't look nice. So it's question, do we kick length indicator from teaser, or change numbers to that teaser looks nice in front-end, or leave it as it is in PR.
2. (length indicator related, but not so relevant) maybe it would be nice to add "change" handler too, because jQuery practice is after setting value for input field -> trigger change event: https://api.jquery.com/val/

**Comment by goldquest** [comment]
Regarding 1: Right now the page description is set to be the field teaser text. Do we maybe need another SEO description field?
Regarding 2: I created an issue for that.

**Comment by dbosen** [comment]
I am also in favor of rethinking the description/teaser field. The length indicator is not working for the description.

**Pull request merged**